backend/src/apiserver/resource/resource_manager.go

```diff
@@ -541,8 +545,20 @@ func (r *ResourceManager) CreateJob(apiJob *api.Job) (*model.Job, error) {
 		return nil, util.Wrap(err, "Create job failed")
 	}
 
-	// Set workflow to be run using default pipeline runner service account.
-	workflow.SetServiceAccount(r.getDefaultSA())
+	multiuserMode := common.IsMultiUserMode()
+	if multiuserMode == true {
+		if len(workflow.Spec.ServiceAccountName) == 0 || workflow.Spec.ServiceAccountName == defaultPipelineRunnerServiceAccount {
+			// To reserve SDK backward compatibility, the backend currently replaces the serviceaccount in multi-user mode.
+			workflow.SetServiceAccount(defaultServiceAccount)
+		}
+	} else {
+		workflow.SetServiceAccount(r.getDefaultSA())
+	}
+
+	// Disable istio sidecar injection
+	if multiuserMode == true {
+		workflow.SetAnnotationsToAllTemplates(util.AnnotationKeyIstioSidecarInject, util.AnnotationValueIstioSidecarInjectDisabled)
+	}
```

Comment:
discussion: these codes are hacky and a duplicate of run api code, would it be worth refactored as a util function

Reply:
discussed through hangout: will have two util functions for different purposes

Reply:
Done